### PR TITLE
Fix: Currencies changing error

### PR DESCRIPTION
**Pull request by lomamech**
Fix: Currencies changing error

### CODE REVIEW

File: client/src/partials/receipts/receipt.sale.js [hunk 1]
```diff
@@ -87,7 +104,6 @@ angular.module('bhima.controllers')
     }
 
     function updateCost(currency_id) {
```
Comment (jniles) [inline client/src/partials/receipts/receipt.sale.js]: Is this function still used?

File: client/src/partials/receipts/receipt.sale.js [hunk 2]
```diff
@@ -67,6 +68,22 @@ angular.module('bhima.controllers')
       }
     };
 
+    function updateCurrency (currency) {
+      model.total.localeCost = model.common.doConvert(model.saleRecords[0].cost, currency, model.saleRecords[0].invoice_date);
```
Comment (jniles) [inline client/src/partials/receipts/receipt.sale.js]: You should really cache this in a local variable.  Like this:
```javascript
var totals = model.total.localCost;
// do something fancy
```
This is both semantic and faster.  The performance increases [are](http://stackoverflow.com/questions/2936807/javascript-object-access-performance) [well](http://moduscreate.com/javascript-performance-tips-tricks/) [documented](http://www.javascriptkit.com/javatutors/efficientjs.shtml).

File: client/src/partials/receipts/receipt.sale.js (hunk 1 above)
Comment (lomamech) [inline client/src/partials/receipts/receipt.sale.js]: Yes, This function is used when downloading page

File: client/src/partials/receipts/receipt.sale.js [hunk 3]
```diff
@@ -67,6 +68,25 @@ angular.module('bhima.controllers')
       }
     };
 
+    function updateCurrency (currency) {
+      var totals = model.total,
+        saleRecords = model.saleRecords[0];
+        
+      totals.localeCost = model.common.doConvert(saleRecords.cost, currency, saleRecords.invoice_date);
```
Comment (jniles) [inline client/src/partials/receipts/receipt.sale.js]: The same logic as totals = model.total can be applied to almost every line of this function.  For example, you call `model.common.doConvert()` potentially 4 + length(model.saleRecords) times in updateCurrency()!  You clearly should cache this function in a local variable that can be called less expensively.

**Pull request merged**